stdlib/public/core/UnsafeRawPointer.swift.gyb

```diff
@@ -0,0 +1,557 @@
+  ///   compatible.
+  /// 
+  /// - Postcondition: The memory is initialized to raw bytes. If the
+  ///   memory is bound to type `U`, then it now contains a value of
+  ///   type `U`.
+  ///
+  /// - Note: A trivial type can be copied with just a bit-for-bit
+  ///   copy without any indirection or reference-counting operations.
+  ///   Generally, native Swift types that do not contain strong or
+  ///   weak references or other forms of indirection are trivial, as
+  ///   are imported C structs and enums.
+  ///
+  /// - Note: Storing a copy of a nontrivial value into memory
+  ///   requires that the user know the type of value previously in
+  ///   memory, and requires initialization or assignment of the
+  ///   memory. This can be acheived via a typed `UnsafeMutablePointer`
+  ///   or via a raw pointer `self`, as follows, where `U` is the
+  ///   previous type and `T` is the copied value's type:
+  ///   `let typedPtr = self.bindMemory(to: U.self, capacity: 1)`
+  ///   `typedPtr.deinitialize(count: 1)`
+  ///   `self.initializeMemory(as: T.self, to: newValue)`
+  public func storeBytes<T>(
+    of value: T, toByteOffset offset: Int = 0, as: T.Type
+  ) {
+    var temp = value
+    withUnsafeMutablePointer(&temp) { source in
+      let rawSrc = UnsafeMutableRawPointer(source)._rawValue
+      Builtin.int_memcpy_RawPointer_RawPointer_Int64(
+        (self + offset)._rawValue, rawSrc, UInt64(sizeof(T.self))._value,
+        /*alignment:*/ Int32(0)._value,
+        /*volatile:*/ false._value)
+    }
+  }
+  
+  /// Copy `count` bytes from `source` into memory at `self`.
+  ///  
+  /// - Precondition: The memory at `source..<source + count` is
+  ///   initialized to some trivial type `T`.
+  ///
+  /// - Precondition: If the memory at `self..<self+count` is bound to
+  ///   a type `U`, then `U` is a trivial type, the underlying
+  ///   pointers `source` and `self` are properly aligned for type
+  ///   `U`, and `count` is a multiple of `strideof(U.self)`.
+  ///
+  /// - Postcondition: The memory at `self..<self+count` is
+  ///   initialized to raw bytes. If the memory is bound to type `U`,
+  ///   then it contains values of type `U`.
+  public func copyBytes(from source: UnsafeRawPointer, count: Int) {
+    _debugPrecondition(
+      count >= 0, "UnsafeMutableRawPointer.copyBytes with negative count")
+    if UnsafeRawPointer(self) < source
+       || (source + count) <= UnsafeRawPointer(self) {
+      // copy forward from a disjoint or following overlapping range.
+      Builtin.int_memcpy_RawPointer_RawPointer_Int64(
+        _rawValue, source._rawValue, UInt64(count)._value,
+        /*alignment:*/ Int32(0)._value,
+        /*volatile:*/ false._value)
+    }
+    else {
+      // copy backward from a non-following overlapping range.
+      var destPtr = self + (count - 1)
+      var srcPtr = source + (count - 1)
+      while destPtr >= self {
+        Builtin.int_memcpy_RawPointer_RawPointer_Int64(
```

Comment:
Could you wrap this builtin in a transparent function, just to add required labels and use non-builtin types?